src/common/init.cpp

```diff
@@ -62,6 +64,35 @@ std::optional<ConfigError> InitConfig(ArgsManager& args, SettingsAbortFn setting
             fs::create_directories(net_path / "wallets");
         }
 
+        // Warn if we are trying to put the datadir on an exFAT fs on MacOS
+        // This is an upstream issue known to cause bugs, see #28552
+#ifdef __APPLE__
+        struct PathCheck {
+            fs::path path;
+            std::string_view description;
+        };
+        std::array<PathCheck, 2> paths_to_check{{
+            {args.GetDataDirNet(), "data directory"},
+            {args.GetBlocksDirPath(), "blocks directory"}
+        }};
```

Comment:
Makes sense to cover blocksdir and datadir.
Since the user could also set `-walletdir=<dir>` when starting bitcoind, do you think it would be worth warning for it as well? Much lighter disk usage than blocksdir and datadir, but if we're doubtful of macOS's exFAT implementation, then maybe we would want to warn the user when storing a wallet with it?